pkg/sqlmigrations/migrations.go

```diff
@@ -512,3 +553,46 @@ func addRootUser(ctx context.Context, r runner) error {
 	}
 	return err
 }
+
+func addRoles(ctx context.Context, r runner) error {
+	// System tables can only be modified by a privileged internal user.
+	session := r.newRootSession(ctx)
+	defer session.Finish(r.sqlExecutor)
+
+	// Add the roles column to the system.users table.
+	const alterStmt = `
+		ALTER TABLE system.users ADD COLUMN IF NOT EXISTS "isRole" BOOL NOT NULL DEFAULT false;
+		 `
+
+	if res, err := r.sqlExecutor.ExecuteStatementsBuffered(
+		session, alterStmt, nil, 1); err == nil {
+		res.Close(ctx)
+	} else {
+		return err
+	}
+
+	// Create the `admin` role.
+	const upsertAdminStmt = `
+		INSERT INTO system.users (username, "hashedPassword", "isRole")
+		  VALUES ($1, '', true);
+						`
+
+	// Add the `admin` role. We retry a few times as the schema change may still be backfilling.
+	pl := tree.MakePlaceholderInfo()
+	pl.SetValue("1", tree.NewDString(sqlbase.AdminRole))
+	var err error
+	for retry := retry.Start(retry.Options{MaxRetries: 5}); retry.Next(); {
+		var res sql.StatementResults
+		res, err = r.sqlExecutor.ExecuteStatementsBuffered(session, upsertAdminStmt, &pl, 1)
+		if err == nil {
+			res.Close(ctx)
+			break
+		}
+		if sqlbase.IsUniquenessConstraintViolationError(err) {
+			log.Fatalf(ctx, `cannot create role %q, a user with that name exists. Please drop the user `+
+				`using a previous version of CockroachDB and try again`, sqlbase.AdminRole)
```

Comment:
Perhaps include the command to run in this message?

Reply:
Done.